model_server/src/model_server/app.py

```diff
@@ -0,0 +1,52 @@
+import asyncio
+from json import JSONDecodeError
+from starlette.applications import Starlette
+from starlette.responses import JSONResponse
+from starlette.routing import Route
+from transformers import pipeline
+
+from model_server import settings
+from model_server.handler import server_loop, DEFAULT, ENCODE, QUESTION_ANSWERING
+
+
+def handler(command):
+    """Make a command specific handler."""
+    async def handle(request):
+        try:
+            payload = await request.json()
+        except JSONDecodeError:
+            return JSONResponse({'error': 'invalid JSON provided'}, status_code=400)
+
+        query = payload.get('query')
+        if not query:
+            return JSONResponse({'error': 'No query provided'}, status_code=400)
+
+        # The model should only be loaded into memory once, seeing it's large.
+        # Use a queue to send/receive messages from it.
+        response_q = asyncio.Queue()
+        await request.app.model_queue.put((command, query, payload, response_q))
+        output, status = await response_q.get()
+        return JSONResponse(output, status_code=status)
+
+    return handle
+
+
+routes = [
+    Route("/", handler(DEFAULT), methods=["POST"]),
+    Route("/encoding", handler(ENCODE), methods=["POST"]),
```

Comment:
These more specific endpoints will cause an error (returns a 400) if they can't support this call.
Now I think about it, it might be better to just have the default endpoint? How many models are there that can support both kinds of actions?